### PR TITLE
Make url check concurrent

### LINKED ISSUE #479: Link checker: make requests in parallel
Given the amount of latency involved in sending a request, it would be nice if the checker performed its requests in parallel.

### PR DESCRIPTION
**Pull request by clample**
Resolves #479

### CODE REVIEW

**Comment by clample** [comment]
I built the example site and added 50 external links from hacker news: [Sample Markdown](https://gist.github.com/clample/00283ed45ed4ac3579e36b481491b6fc)

I ran the link checker against the sample site to see if these changes actually improved performance. I ran the link check 10 times with the concurrent link check and 10 times with master. The average time improved from 27.2 seconds to 20.5 seconds.

Actual results:

With concurrency:

13.882
13.807
16.456
13.639
14.385
15.356
20.795
28.657
40.384
27.241

average: 20.4602

Master (without concurrency):

17.040
30.740
29.492
32.440
27.697
25.157
26.285
30.854
23.844
28.244

average: 27.1793

**Comment by clample** [comment]
I might also have time to add tests (we'll see)

File: src/Hakyll/Check.hs [hunk 1]
```diff
@@ -191,58 +233,47 @@ checkInternalUrl base url = case url' of
...
+                _ -> faulty url (Just $ showException e)
+        Right _ -> ok url
+    where
+        -- Convert exception to a concise form
+        showException e = case cast e of
+            Just (Http.HttpExceptionRequest _ e') -> show e'
+            _ -> head $ words $ show e
 #else
-checkExternalUrl _ = return ()
+checkExternalUrl url = skip url Nothing
```
Comment (clample) [inline src/Hakyll/Check.hs]: We do `skip` to make sure we eventually write to the MVar. Otherwise I believe that there's an error

**Comment by jaspervdj** [comment]
Looks good to me 👍 (although I think this could be simplified a bit by just using an `IORef` instead of an `MVar`).

Can I merge this in or do you want to add additional work?

**Comment by clample** [comment]
Cool! (I didn't know what an `IORef` is but I'll check it out).

You're welcome to merge this

**Pull request merged**